app/assets/javascripts/exercises.js

```diff
@@ -87,12 +87,11 @@ $(document).on('turbolinks:load', function () {
         // validate fileUrl
         const matches = fileUrl.match(/files\/(\d+)/);
         if (matches) {
-            // select the file form based on the delete button
-            const fileForm = $(`*[data-file-url="${fileUrl}"]`).parent().parent().parent();
-            fileForm.remove();
+            const fileId = matches[1];
+
+            $(`#file-form-${fileId}`).remove();
 
             // now remove the hidden input representing the file
```

Comment:
As the previous comment has been deleted, this comment seems somewhat "incomplete". I would either keep (and update) both or remove them.

Reply:
I thought this comment has come value because the need to remove the hidden input was not apparent to me. I rewrote it a little to make it clearer.

The pervious comment was removed because I thought it is apparent what the code is doing.